### PR TITLE
Fix start_author in get_discussions_by_comments. #606

### PR DESCRIPTION
**Pull request by afalaleev**

### CODE REVIEW

File: plugins/social_network/social_network.cpp [hunk 1]
```diff
@@ -790,15 +802,38 @@ namespace golos { namespace plugins { namespace social_network {
...
 
         auto& db = pimpl->database();
         return db.with_weak_read_lock([&]() {
-            return pimpl->select_ordered_discussions<sort::by_updated>(
-                query,
-                [&](const discussion& d) -> bool {
-                    return true;
+            const auto &idx = db.get_index<comment_index>().indices().get<by_author_last_update>();
+            auto itr = idx.lower_bound(*query.start_author);
+            FC_ASSERT(itr != idx.end(), "Comment is not in account's comments");
```
Comment (zxcat) [inline plugins/social_network/social_network.cpp]: Assert message looks wrong, here is author check, not comment.

File: plugins/social_network/social_network.cpp [hunk 2]
```diff
@@ -790,15 +802,38 @@ namespace golos { namespace plugins { namespace social_network {
...
+
+            if (!!query.start_permlink) {
+                const auto &lidx = db.get_index<comment_index>().indices().get<by_permlink>();
+                auto litr = lidx.find(boost::make_tuple(*query.start_author, *query.start_permlink));
```
Comment (zxcat) [inline plugins/social_network/social_network.cpp]: `std::make_tuple` will be better

File: plugins/social_network/social_network.cpp (hunk 1 above)
Comment (afalaleev) [reply plugins/social_network/social_network.cpp]: Fixed.

File: plugins/social_network/social_network.cpp (hunk 2 above)
Comment (afalaleev) [reply plugins/social_network/social_network.cpp]: Fixed.

**Pull request merged**